syntax = "proto3";

package book_database;

message GetBookRequest {
  string request_id = 1;
}

message Empty {}

message Book {
  string id = 1;
  string title = 2;
  string author = 3;
  string description = 4;
  int32 copies= 5;
  int32 copiesAvailable = 6;
  string category = 7;
  string img = 8;
  float price = 9;
}

message BookList {
  repeated Book books = 1;
}

service BookDatabaseService {
  rpc AddBook(Book) returns (Book);
  rpc GetBook(GetBookRequest) returns (Book);
  rpc ListBooks(Empty) returns (BookList);
  rpc UpdateBook(Book) returns (Book);
  rpc DeleteBook(GetBookRequest) returns (Empty);
}